habitat_sim/utils/datasets_download.py

```diff
@@ -0,0 +1,246 @@
+        return
+    print(
+        "Cleaning datasource " + uid + ' directory: "' + data_sources[uid]["root"] + '"'
+    )
+    os.system("rm -r " + data_sources[uid]["root"])
+
+
+def download_and_place(uid, replace=False):
+    if uid not in data_sources:
+        print("Data download failed, no datasource named " + uid)
+        return
+
+    # check for current version
+    version_filepath = os.path.join(
+        data_dir,
+        data_sources[uid]["root"] + "DATA_VERSION_" + data_sources[uid]["version"],
+    )
+    if os.path.exists(data_sources[uid]["root"]):
+        resolved_existing = False
+        replace_existing = False
+        for file in os.listdir(data_sources[uid]["root"]):
+            if "DATA_VERSION" in file:
+                existing_version = file[13:]
+                if existing_version == data_sources[uid]["version"]:
+                    print(
+                        "Existing data source ("
+                        + uid
+                        + ") version ("
+                        + existing_version
+                        + ") is current, aborting download."
+                        + " Root directory: "
+                        + data_sources[uid]["root"]
+                    )
+                    return
+                else:
+                    # found root directory with different version
+                    found_version_message = (
+                        "("
+                        + uid
+                        + ") Found previous data version ("
+                        + existing_version
+                        + "). Replace with requested version ("
+                        + data_sources[uid]["version"]
+                        + ")?"
+                    )
+                    replace_existing = (
+                        replace if replace else prompt_yes_no(found_version_message)
+                    )
+                    resolved_existing = True
+        if not resolved_existing:
+            # found root directory with unspecified version
+            found_version_message = (
+                "("
+                + uid
+                + ") Found unknown data version. Replace with requested version ("
+                + data_sources[uid]["version"]
+                + ")?"
+            )
+            replace_existing = (
+                replace if replace else prompt_yes_no(found_version_message)
+            )
+        if replace_existing:
+            clean_data(uid)
+        else:
+            print("Not replacing data, aborting download.")
+            return
+
+    # download
+    download_pre_args = (
+        ""
+        if "download_pre_args" not in data_sources[uid]
+        else data_sources[uid]["download_pre_args"]
+    )
+    download_post_args = (
+        ""
+        if "download_post_args" not in data_sources[uid]
+        else data_sources[uid]["download_post_args"]
+    )
+    download_command = (
+        "wget --continue "
```

Comment:
Why call wget? Pythonhas plenty of ways to handle downloads itself like requests. No reason to add an dependency like this. (And no properly shell escape it).

Reply:
Requests or url lib should get you what you need here iwth better error handling

Reply:
```
import urllib.request
import shutil
...
# Download the file from `url` and save it locally under `file_name`:
with urllib.request.urlopen(url) as response, open(file_name, 'wb') as out_file:
    shutil.copyfileobj(response, out_file)
```

Reply:
My 2 cents here is that `wget` is worth it.  The ability to resume partial downloads can be important when on flaky internet connections.

Reply:
No external dependency needed for this: @erikwijmans  https://gist.github.com/tobiasraabe/58adee67de619ce621464c1a6511d7d9